Dockerfile

```diff
@@ -0,0 +1,31 @@
+# The builder image, used to build the virtual environment
+FROM python:3.11-buster as builder
+
+RUN apt-get update && apt-get install -y git
+
+RUN pip install poetry==1.4.2
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+# A directory to have app data 
+WORKDIR /app
+
+COPY pyproject.toml poetry.lock ./
+
+RUN poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR
+
+# The runtime image, used to just run the code provided its virtual environment
+FROM python:3.11-slim-buster as runtime
+
+ENV VIRTUAL_ENV=/app/.venv \
+    PATH="/app/.venv/bin:$PATH"
+
+COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+
+COPY ./streamlit_agent ./streamlit_agent
+
+
+CMD ["streamlit", "run", "streamlit_agent/chat_pandas_df.py", "--server.port", "8080"]
```

Comment:
Should this port be 8051 instead of 8080? or else change the line below that uses 8051 `docker run -d --name langchain-streamlit-agent -p 8051:8051 langchain-streamlit-agent:latest`

Reply:
I have changed to 8051 in `Dockerfile`.